database/pgx/pgx.go

```diff
@@ -211,11 +222,20 @@ func (p *Postgres) Open(url string) (database.Driver, error) {
 }
 
 func (p *Postgres) Close() error {
-	connErr := p.conn.Close()
-	dbErr := p.db.Close()
-	if connErr != nil || dbErr != nil {
-		return fmt.Errorf("conn: %v, db: %v", connErr, dbErr)
+	if p.stdlib != nil {
```

Comment:
We'll need to revisit this once the fields are renamed. The current field names make this logic hard to follow.

The `pgx.Conn` should be closed first and only released if the stdlib was used.

Reply:
I'm under the impression that if we got it from AquireConn we're just "borowing" it from the sql.DB, so we shouldn't be closing it, but passing it back to the sql.DB instance with ReleaseConn, which is why I did it this way.

[The docs for AcquireConn](https://pkg.go.dev/github.com/jackc/pgx/v4/stdlib#AcquireConn) say: 
> AcquireConn acquires a *pgx.Conn from database/sql connection pool. It must be released with ReleaseConn.

Reply:
Ahhh, that makes sense. e.g. if the connection is create/acquired via `AcquireConn()`, it should be returned/released via `ReleaseConn()` and not close. Similarly, [`database.sql.Conn.Raw()`](https://github.com/golang/go/blob/go1.16.5/src/database/sql/sql.go#L1929) doesn't close connections and releases it when finished.